client/src/document/ingredients/media-feature-section.tsx

```diff
@@ -0,0 +1,123 @@
+function getMediaQueryValue(featureName, valueType, enumValues) {
+  switch (valueType.toLowerCase()) {
+    case "enum":
+      return getEnumValue(featureName, enumValues);
+    case "integer":
+      return getNumericalValue(featureName, 0, "");
+    case "length":
+      return getNumericalValue(featureName, 0, "px");
+    case "resolution":
+      return getNumericalValue(featureName, 1, "dpi");
+    case "ratio":
+      if (featureName === "aspect-ratio") {
+        return getAspectRatio(
+          featureName,
+          window.innerWidth,
+          window.innerHeight
+        );
+      } else if (featureName === "device-aspect-ratio") {
+        return getAspectRatio(
+          featureName,
+          window.screen.width,
+          window.screen.height
+        );
+      }
+  }
+
+  throw new Error(
+    '"' +
+      featureName +
+      '" media feature value couldn\'t be calculated, because value type "' +
+      valueType +
+      '" is not supported.'
+  );
+
+  function testValue(featureName, value) {
+    return window.matchMedia("(" + featureName + ": " + value + ")").matches;
+  }
+
+  function getEnumValue(featureName, enumValues) {
+    for (let value of enumValues) {
+      if (value !== undefined && testValue(featureName, value)) return value;
+    }
+    return undefined; //Feature is not supported or array of enum values is incorrect.
+  }
+
+  function getNumericalValue(featureName, minimumPossibleValue, unit) {
+    if (!testValue("min-" + featureName, minimumPossibleValue + unit))
+      return undefined; //Feature is not supported
+
+    let reminder;
+    let lower = 0;
+    let higher = 8192;
+    let triesLeft = 20;
+    let value = parseInt(String((lower + higher) / 2));
+
+    while (triesLeft > 0 && value - lower > 1 && higher - value > 1) {
+      triesLeft--;
+      if (testValue("min-" + featureName, value + unit)) {
+        lower = value;
+        reminder = 0;
+      } else {
+        higher = value;
+        reminder = 1;
+      }
+      value = parseInt(String((higher + lower) / 2));
+    }
+    value -= reminder;
+    return value + unit;
+  }
```

Comment:
This didn't work for `{{MediaFeatureValue("color", "integer")}}`. Please add a comment explaining what your doing, not everyone just spots that this is a binary search. Also adding template strings helps, I think. I'd recommend some adjustments:
```js
  function getNumericalValue(featureName, minimumPossibleValue, unit) {
    if (!testValue(`min-${featureName}`, `${minimumPossibleValue}${unit}`))
      return undefined; //Feature is not supported

    let lower = 0;
    let higher = 8192;
    let triesLeft = 20;
    let value = Math.floor((lower + higher) / 2);

    while (triesLeft > 0 && higher - lower > 1) {
      triesLeft--;
      if (testValue(`min-${featureName}`, `${value}${unit}`)) {
        lower = value;
      } else {
        higher = value;
      }
      value = Math.floor((higher + lower) / 2);
    }
    return `${value}${unit}`;
  }
```